File: cpp/1108.defanging-an-ip-address.cpp
/*
 * @lc app=leetcode id=1108 lang=cpp
 *
 * [1108] Defanging an IP Address
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    string defangIPaddr(string address)
    {
        string res;
        for (auto c : address)
        {
            if (c != '.')
            {
                res.push_back(c);
            }
            else
            {
                res.append("[.]");
            }
        }
        return res;
    }
};
// @lc code=end

File: 1108.defanging-an-ip-address.java

import java.util.ArrayList;

/*
 * @lc app=leetcode id=1108 lang=java
 *
 * [1108] Defanging an IP Address
 */

// @lc code=start
import java.text.CharacterIterator;
import java.text.StringCharacterIterator;
class Solution {
    public String defangIPaddr(String address) {
        StringBuilder res = new StringBuilder();
        var iter = new StringCharacterIterator(address);
        while (iter.current() != CharacterIterator.DONE) {
            if (iter.current() != '.') {
                res.append(iter.current());
            } else {
                res.append("[.]");
            }
            iter.next();
        }
        return res.toString();
    }
}
// @lc code=end
